site/public/js/notebook_builder/form-options-widget.js

```diff
@@ -0,0 +1,72 @@
+class FormOptionsWidget extends Widget {
+    constructor() {
+        super();
+    }
+
+    render() {
+        const heading_container = this.getHeadingContainer('Form Options');
+
+        const interactive_container = this.getInteractiveContainer();
+        interactive_container.innerHTML = this.getFormOptionsTemplate();
+
+        const container = document.createElement('div');
+        container.classList.add('notebook-builder-widget');
+        container.classList.add('form-options-widget');
+        container.appendChild(heading_container);
+        container.appendChild(interactive_container);
+
+        const save_button = container.querySelector('.save-button');
+        save_button.addEventListener('click', event => {
+            this.saveButtonAction();
+        })
+
+        return container;
+    }
+
+    /**
+     * Get the template used in the form options widget
+     *
+     * @returns {string}
+     */
+    getFormOptionsTemplate() {
+        return `
+        <div class="buttons">
+            <input type="button" class="save-button" value="Save">
+        </div>
+        <div class="status"></div>`;
+    }
+
+    /**
+     * Pack up all the configuration data from the GUI into a nice config.json and submit it to the server
+     */
+    saveButtonAction() {
+        const file = new File([JSON.stringify(notebook_builder.getJSON(), null, 2)], 'config.json', {type: "text/plain"});
+        const g_id = window.location.pathname.split('/').pop();
+        const url = buildCourseUrl(['notebook_builder', 'save']);
+
+        const form_data = new FormData();
+        form_data.append('config_upload', file, 'config.json');
+        form_data.append('csrf_token', csrfToken);
+        form_data.append('g_id', g_id);
+
+        const makeRequest = async () => {
```

Comment:
is the `makeRequest` function used more than once? If not probably don't need to put the request in a function.

Reply:
`await` only works inside of `async` functions.